app/src/main/java/com/android/sample/ui/account/EditAccountScreen.kt

```diff
@@ -0,0 +1,451 @@
+                .size(50.dp)
+                .offset(x = (-7.5).dp, y = (-7.5).dp)
+                .background(MaterialTheme.colorScheme.background, shape = CircleShape)
+                .border(1.dp, Color.Black, CircleShape)
+                .clickable { openGallery(photoPickerLauncher) }) {
+          Icon(
+              imageVector = Icons.Default.CameraAlt,
+              contentDescription = CHANGE_PROFILE_PICTURE_BUTTON_DESCRIPTION,
+              modifier =
+                  Modifier.fillMaxSize().padding(5.dp).testTag(CHANGE_PROFILE_PICTURE_BUTTON_TAG),
+              tint = Color.Black)
+        }
+  }
+}
+
+/**
+ * Function that creates an input text field composable for a given field
+ *
+ * @param boxName the name given to the text field, which will be displayed above the value inside
+ *   it
+ * @param boxValue the value which will be contained in the text field
+ * @param readOnly [Boolean] indicating if the field is read only or not
+ * @param onValueChange action to be performed when the value of the field changes
+ */
+@Composable
+private fun InputTextBox(
+    boxName: String,
+    boxValue: String,
+    readOnly: Boolean,
+    onValueChange: (String) -> Unit
+) {
+  OutlinedTextField(
+      value = boxValue,
+      onValueChange = onValueChange,
+      modifier = Modifier.width(350.dp).testTag("$boxName text field"),
+      textStyle = TextStyle(fontSize = 16.sp),
+      label = { Text(boxName) },
+      readOnly = readOnly,
+      singleLine = true)
+}
+
+/**
+ * Function that creates the composable which contains the date picker
+ *
+ * @param newDateOfBirth the date of birth which will be displayed and stored in the future
+ * @param onValueChange action to be performed when the value of date of birth changes
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun DateOfBirthBox(newDateOfBirth: String?, onValueChange: (String) -> Unit) {
+  var showDatePicker by remember { mutableStateOf(false) }
+  val datePickerState = rememberDatePickerState()
+  var displayedDate by remember { mutableStateOf(newDateOfBirth!!) }
+  val selectedDate =
+      datePickerState.selectedDateMillis?.let {
+        val formatter = SimpleDateFormat("dd/MM/yyyy", Locale.getDefault())
+        formatter.format(Date(it))
+      } ?: " "
+
+  OutlinedTextField(
+      value = displayedDate,
+      onValueChange = {},
+      textStyle = TextStyle(fontSize = 16.sp),
+      label = { Text(stringResource(R.string.date_of_birth_label)) },
+      readOnly = true,
+      trailingIcon = {
+        IconButton(
+            modifier = Modifier.testTag(DATE_OF_BIRTH_CHANGE_BUTTON_TAG),
+            onClick = { showDatePicker = !showDatePicker }) {
+              Icon(
+                  imageVector = Icons.Default.DateRange,
+                  contentDescription = DATE_OF_BIRTH_FIELD_DESCRIPTION)
+            }
+      },
+      modifier = Modifier.width(350.dp).testTag(DATE_OF_BIRTH_TEXT_FIELD_TAG))
```

Comment:
I feel like you could use the InputTextBox instead of a custom outlined text field to increase readability and avoid duplicated code. You would just need to add a trailing Icon parameter that can be a unit by default. and the testtag would need to be changed in the tests as the InputTextBox generates it's own.

To be honest, I don't think it's that necessary, but because I have some corrections for the InputTextBox class, to make sure they're also done here it would be a good idea to just reuse it here.

Reply:
Because it did not take a lot of effort to make the changes, I did it. Take a look and tell me what do you think about the changes.